dj_hetmech_app/views.py

```diff
@@ -26,45 +22,47 @@ class NodeView(ModelViewSet):
     search_fields = ('identifier', 'metanode__identifier', 'name')
 
 
-class PathCountView(ModelViewSet):
+class QueryPairView(APIView):
     http_method_names = ['get']
-    serializer_class = PathCountSerializer
 
-    def get_queryset(self):
-        """Optionally restricts the returned PathCount results to a given
-        user by filtering against the two query parameters in the URL.
+    def polish_pathcounts(self, source_id, target_id, pathcounts_data):
+        """Polish pathcounts data to make source/target consistent with
+        user queries.
         """
-        queryset = PathCount.objects.all()
-        node1 = self.request.query_params.get('qsource', None)
-        node2 = self.request.query_params.get('qtarget', None)
-        if node1 and node2:
-            queryset = queryset.filter(
-                Q(source=node1, target=node2) | Q(source=node2, target=node1)
-            )
-
-        return queryset
-
 
-class QueryPairView(APIView):
-    http_method_names = ['get']
+        for entry in pathcounts_data:
+            # If necessary, swap "source_degree" and "target_degree"
+            # values in each pathcount entry's "dgp".
+            if (int(source_id) != entry['source']):
```

Comment:
we can remove parenthesis in the if statement line above.

Reply:
@dongbohu we could retain the info on whether reversal was performed:

```python
reversed = int(source_id) != entry['source']
entry['reversed'] = reversed
if reversed:
```

This could be useful for some applications perhaps and could help address any user confusion over the reversal?

Reply:
Sure!